.github/workflows/artifacts.yml

```diff
@@ -0,0 +1,83 @@
+name: Artifacts generation and upload
+on:
+  workflow_run:
+    workflows: ["Mattermost Build"]
+    types:
+      - completed
+jobs:
+  upload-s3:
+    name: Upload artifacts to S3
+    runs-on: ubuntu-latest
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: cd/Configure AWS
+        uses: aws-actions/configure-aws-credentials@07c2f971bac433df982ccc261983ae443861db49 # v1-node16
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.PR_BUILDS_BUCKET_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.PR_BUILDS_BUCKET_AWS_SECRET_ACCESS_KEY }}
+      - name: cd/Download artifacts
+        uses: dawidd6/action-download-artifact@0c49384d39ceb023b8040f480a25596fd6cf441b # v2.26.0
+        with:
+          workflow_conclusion: success
+          name: server-dist-artifact
+      # Get Branch name from calling workflow
+      # Search for the string "pull" and replace it with "PR" in branch-name
+      - name: cd/Get branch name
```

Comment:
What if we make this a reusable one in `actions` and use it from there? I see lots of benefits of it as I have seen other repos using it.

Reply:
This is actually the only pipeline using this. 
I am finding this an overkill. It is an oneliner. It only performsa  simple text substitution ( Search for the string "pull" and replace it with "PR" in branch-name). I'd expect a seperate action, to be introduced and except reusability, to perform some amount of work. 

Except that, the Branch variable, is changing depending on the usage. 
In this case, it is `github.event.workflow_run.head_branch` which refers to the Branch name from calling workflow. We could input this as a variable to a newly created action, but I really find this as introducing complexity for a simple text substitution ( Search for the string "pull" and replace it with "PR" in branch-name) . To be honest, not sure if this is needed. 

I have a feeling that this maybe tech dept introduced with the migration from CircleCI 🤷

Reply:
I remember webapp was using it too, right? Other than that I agree with you

Reply:
Isn't webapp now part of this repo?